sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/physical/partitioning.scala

```diff
@@ -951,10 +951,11 @@ case class KeyGroupedShuffleSpec(
       }
 
   override def createPartitioning(clustering: Seq[Expression]): Partitioning = {
-    val newExpressions: Seq[Expression] = clustering.zip(partitioning.expressions).map {
-      case (c, e: TransformExpression) => TransformExpression(
-        e.function, Seq(c), e.numBucketsOpt)
-      case (c, _) => c
+    val clusteringMap = distribution.clustering.map(_.canonicalized).zip(clustering).toMap
```

Comment:
Should we assert the size of `distribution.clustering` matches `clustering`? Or they are definitely matched already.